code/modules/integrated_electronics/core/integrated_circuit.dm

```diff
@@ -80,23 +80,21 @@ a creative player the means to solve many problems.  Circuits are held inside an
 		return user.canUseTopic(src, BE_CLOSE)
 
 /obj/item/integrated_circuit/Initialize()
-	. = ..()
```

Comment:
IIRC this was put here for a reason. I think circuit imprinters broke if it wasn't here.

See #787

Reply:
I'll see to it later on.

Reply:
I dove a little bit through the code and noticed what's wrong: it's the way the printer handles iron. It reads the iron from a material list as a path. However, most items' materials have iron initialized as "iron" = number inside the list and use that "iron" variable to recycle through the get_item_material_amount() proc.

Also: the .=..() calls obj/item/Initialize(), which sets the material depending on the item's material instead of the cost so it needs to be put first. It's just a question of adjusting the printer's building with the get_item_material_amount() proc, which I'll see to soon.